### PR TITLE
[path_provider_linux] Migration to null safety

### PR DESCRIPTION
**Pull request by yash1200**
## Description

Migrates `path_provider_linux` to null safety.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/70219

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by yash1200** [comment]
Hey @ditman kindley have a look at this one.

**Review by passsy — APPROVED** [review]

**Review by Hixie — APPROVED** [review]
LGTM though it looks like the analysis isn't happy.

**Comment by yash1200** [comment]
Analyze is not getting passed because there is no environment set in connectivity plugin's pubspec.yaml, should I add that in this PR?

**Comment by yash1200** [comment]
@Hixie I think the analyze test errors are not related to this PR.

File: packages/path_provider/path_provider_linux/pubspec.yaml [hunk 1]
```diff
@@ -11,17 +11,17 @@ flutter:
...
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.12.0-0 <3.0.0"
   flutter: ">=1.10.0 <2.0.0"
 
 dependencies:
-  path: ^1.6.4
-  xdg_directories: ^0.1.0
-  path_provider_platform_interface: ^1.0.1
+  path: ^1.8.0-nullsafety.3
+  xdg_directories: ^0.1.2
```
Comment (yash1200) [inline packages/path_provider/path_provider_linux/pubspec.yaml:19]: The PR to migrate xdg_directories to nnbd is here https://github.com/flutter/packages/pull/250

**Comment by blasten** [comment]
Sorry for the late notice. The nnbd branch has been merged into master. 
You might need to reopen this PR.

**Comment by yash1200** [comment]
Sure I'll reopen the PR on master branch.

**Pull request closed without merging**